lib/internal/modules/esm/get_format.js

```diff
@@ -66,6 +72,16 @@ function extname(url) {
   return '';
 }
 
+/**
+ * Determine whether the given file URL is under a `node_modules` folder.
+ * This function assumes that the input has already been verified to be a `file:` URL,
+ * and is a file rather than a folder.
+ * @param {URL} url
+ */
+function underNodeModules(url) {
+  return StringPrototypeIndexOf(url.pathname, '/node_modules/') !== -1;
+}
```

Comment:
While there are still various edge cases here, overall this seems like the right kind of tradeoff to me, especially considering this will be an opt-in experimental flag.

If there are any concerns about cases where developers happen to be nesting node_modules for whatever reason, I just wanted to note that there could be a slightly more constrained version of this check based on the following rules:

* Apply getPackageBase to the `url` to get the package.json base
* If that package.json file is not the base of a package - ie it is not the `node_modules/pkg/package.json` or `node_modules/@pkg/scope/package.json` file, then we could allow switching that default, so that an inner package.json might also be covered by the switch

Could be worth considering depending on how the feedback goes.

Reply:
This was the code that GitHub Copilot suggested. I felt like it couldn’t possibly be this simple, but I kept looking at it and thinking “what am I missing here” and eventually was like, well, maybe this _is_ all we need. I still feel like there’s some edge case I’m not considering, and that this function will grow (and that’s a reason why it’s a separate function in the first place, even though it’s a one-liner) so if you can think of any cases that this logic doesn’t cover, please let me know.

I don’t think there’s much value in having the flag apply to certain subtrees under `node_modules`. That seems to me like a clear boundary between the user’s app and their dependencies, and it would be frustrating for the flag to apply to dependencies. I’m not sure if this is what you mean by “nesting `node_modules`“?

Reply:
Okay, let's start with this version, and iterate with feedback if we need to.